File: src/test/java/org/code4everything/hutool/UtilsTest.java
package org.code4everything.hutool;

import cn.hutool.core.util.RandomUtil;
import javassist.ClassPool;
import javassist.CtClass;
import javassist.CtMethod;
import javassist.NotFoundException;
import org.junit.Test;

/**
 * @author pantao
 * @since 2021/7/2
 */
public class UtilsTest {

    @Test
    public void getMethodFullInfo() throws NotFoundException {
        ClassPool classPool = ClassPool.getDefault();
        CtClass ctClass = classPool.getCtClass(RandomUtil.class.getName());
        CtMethod ctMethod = ctClass.getDeclaredMethod("randomLong", new CtClass[]{classPool.get(long.class.getName()), classPool.get(long.class.getName())});
        Utils.getMethodFullInfo(ctMethod, null);
    }
}

File: src/test/java/org/code4everything/hutool/UtilsTest.kt
package org.code4everything.hutool

import cn.hutool.core.util.RandomUtil
import javassist.ClassPool
import org.junit.Test

/**
 * @author pantao
 * @since 2021/7/2
 */
class UtilsTest {

    @Test
    fun getMethodFullInfo() {
        ClassPool.getDefault().run { getCtClass(RandomUtil::class.java.name) }.run {
            var longClazz = classPool[Long::class.javaPrimitiveType!!.name]
            getDeclaredMethod("randomLong", Array(2) { longClazz })
        }.let { Utils.getMethodFullInfo(it, null) }.also(::println)
    }
}
